tools/cli/util.go

```diff
@@ -729,6 +729,15 @@ func newContextForLongPoll(c *cli.Context) (context.Context, context.CancelFunc)
 	return newContextWithTimeout(c, defaultContextTimeoutForLongPoll)
 }
 
+func newContextForBackground(c *cli.Context) (context.Context, context.CancelFunc) {
```

Comment:
What does word "Bacground" mean here?

Reply:
it basically follows the context API, that defines Background as 
```
// Background returns a non-nil, empty Context. It is never canceled, has no
// values, and has no deadline. It is typically used by the main function,
// initialization, and tests, and as the top-level Context for incoming
// requests.
```

Reply:
On the second thought, `Background` is probably not the best name since the context that it creates does actually have cancellation from `WithCancel` decorator. Also it may have a timeout if `--context-timeout` flag is passed. 
Taking this into account, another name can be `ForWatch` implying that this context is good for commands that run indefinitely ?
cc @samarabbas

Reply:
What's about `newIndefiniteContext`? Name should reflect what func does, not how the func is implemented. `context.Background()` actually means "take default background context".

Reply:
@alexshtin this context will actually timeout if `--context-timeout` flag is passed.
This is how context will behave:
 - run _indefinitely_ until the workflow is completed
 - run until _timeouts_ IF `--context-timeout` is passed

Reply:
I think the best way is to use func bellow then. Just modify `NewContextWithTimeoutAndCLIHeaders` to accept 0 or negative timeout with will be treated as "no timeout".

Reply:
I tried that and the behavior i'm seeing is that context immediately times out when you pass 0.
According to Maxim that is the expected behavior

Reply:
renamed the context to IndefiniteContext